### DISCUSSION TITLE
osu!lazer deadlock occuring when switching ingame resolution

### DISCUSSION THREAD

**Discussion by i2021** [ppy/osu#21410] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Wanted to squeeze out more preformance from osu!lazer by decreasing ingame resolution, ended up being almoust deadlocked out of the game.

After switching resolution osu instantly minimizes to taskbar and upon reopening it instantly minimizes once again.
You can once again access osu by alt tabbing and spamming F11 to switch mode to windowed/borderless. Issue is that you cannot reverse resolution to previous setting thus being forced to play in borderless/windowed mode only.

### Screenshots or videos

(Video shows black screen for much longer than it is in reality)

https://user-images.githubusercontent.com/72861096/203868478-8253af1c-3c7d-4a21-979c-8599e301115e.mp4

### Version

2022.1123.0

### Logs

[network.log](https://github.com/ppy/osu/files/10087492/network.log)
[performance.log](https://github.com/ppy/osu/files/10087493/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/10087494/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/10087495/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/10087496/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/10087497/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/10087498/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10087499/updater.log)
[database.log](https://github.com/ppy/osu/files/10087500/database.log)
[input.log](https://github.com/ppy/osu/files/10087501/input.log)

**Comment by Susko3** [comment, marked as answer]
You can delete `framework.ini` in `%appdata%\osu` to reset the resolution (or delete the `SizeFullscreen = ` line) as a temporary fix.

Could you show a video of the window automatically minimising?

I would be helpful to get a clean log file of the problem. Could you post the log files after doing the following:
1. Use the default resolution in fullscreen
2. Change to a different resolution
3. Try to focus the game (by alt-tab or otherwise)

Please leave a 10-second gap in-between the steps, so it's easily visible in the log file.

**Reply by i2021** [reply]
https://user-images.githubusercontent.com/72861096/203870194-d175fa42-c86f-4695-bf03-69200bb5a04c.MOV

**Reply by Susko3** [reply]
The log files don't show any errors. Just the expected:
```cs
2022-11-24 22:24:14 [verbose]: Updated display mode to fullscreen resolution: 1366x768@120, SDL_PIXELFORMAT_RGB888
```

Interesting frame from just after the black screen:

![image](https://user-images.githubusercontent.com/16479013/203872108-cb7c6d06-7fda-48cf-bdf3-f8a0fdae534c.png)

**Reply by i2021** [reply]
Soo, is this an issue that can be replicated on other systems or is it just limited to my laptop?

**Comment by i2021** [comment]
[input.log](https://github.com/ppy/osu/files/10087544/input.log)
[network.log](https://github.com/ppy/osu/files/10087545/network.log)
[performance.log](https://github.com/ppy/osu/files/10087546/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/10087547/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/10087548/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/10087549/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/10087550/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/10087551/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10087552/updater.log)
[database.log](https://github.com/ppy/osu/files/10087553/database.log)
